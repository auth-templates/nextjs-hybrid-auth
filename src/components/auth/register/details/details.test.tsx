import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Details from './details';

describe('Details', () => {
    
    test('if all elements are present', () => {
        render(<Details />);
        expect(screen.getByLabelText('First name (optional)')).toBeInTheDocument();
        expect(screen.getByLabelText('Last name (optional)')).toBeInTheDocument();
        expect(screen.getByLabelText('Company (optional)')).toBeInTheDocument();
    })

    it('renders correctly', () => {
        const tree = renderer.create(<Details />).toJSON();
        expect(tree).toMatchSnapshot();
    })
})