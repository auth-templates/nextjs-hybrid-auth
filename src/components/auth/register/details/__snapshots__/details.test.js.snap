// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Details renders correctly 1`] = `
<div
  className="details"
>
  <div
    className="group"
  >
    <label
      htmlFor="firstname"
    >
      First name (optional)
    </label>
    <input
      className="normal"
      id="firstname"
      name="firstname"
    />
  </div>
  <div
    className="group"
  >
    <label
      htmlFor="lastname"
    >
      Last name (optional)
    </label>
    <input
      className="normal"
      id="lastname"
      name="lastname"
    />
  </div>
  <div
    className="group"
  >
    <label
      htmlFor="company"
    >
      Company (optional)
    </label>
    <input
      className="normal"
      id="company"
      name="company"
    />
  </div>
</div>
`;
